File: Dietas/Sistema_Planejamento_Dietas_Refeicoes/Models/Refeicao.cs
using System;

namespace Sistema_Planejamento_Dietas_Refeicoes.Models;

public class Refeicao
{
    public int Id { get; set; }
    public string? Nome { get; set; }
    public string? Descricao { get; set; }
    public DateTime DataRefeicao { get; set; }
    public int UsuarioId { get; set; }
    public Usuario? Usuario { get; set; }
    public List<RefeicaoAlimento> RefeicaoAlimentos { get; set; } = new List<RefeicaoAlimento>();
}

File: front/src/types/Refeicao.ts
import Alimento from "./Alimento";
import { Usuario } from "./Usuario";

export interface RefeicaoAlimento {
  alimentoId: number;
  quantidade: number; // g ou ml
  alimento?: Alimento; // Opcional, usado apenas para exibir o nome na tela
}

export interface Refeicao {
  id?: number;
  usuarioId: number;
  nome: string;
  descricao?: string;
  dataRefeicao: string; // ISO string (YYYY-MM-DDTHH:mm)
  refeicaoAlimentos: RefeicaoAlimento[];
  usuario?: Usuario; // Opcional, para exibição
}